src/vanilla/utils/atomWithLazy.ts

```diff
@@ -0,0 +1,12 @@
+import { PrimitiveAtom, atom } from '../../vanilla.ts'
+
+export function atomWithLazy<Value>(
+  makeInitial: () => Value,
+): PrimitiveAtom<Value> {
+  return {
+    ...atom(undefined as unknown as Value),
+    get init() {
+      return makeInitial()
```

Comment:
should this function be called only once crossing the store?

Reply:
Yes, it should.

Reply:
> should this function be called only once crossing the store?

No, it should call once for each store.
https://github.com/pmndrs/jotai/issues/2458#issuecomment-2024163904

Reply:
Oh, I thought that is what they meant. Once per each store is right.